src/QMCDrivers/MCCoords.hpp

```diff
@@ -0,0 +1,118 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MCCOORDS_HPP
+#define QMCPLUSPLUS_MCCOORDS_HPP
+
+#include "Configuration.h"
+#include "type_traits/complex_help.hpp"
+#include "ParticleBase/RandomSeqGenerator.h"
+
+#include <vector>
+
+namespace qmcplusplus
+{
+
+enum class MCCoordsTypes
```

Comment:
Yes. we need this enum. `CoordsTypes` no need of MC.
Prefer longer names. So use `POS` and `POS_SPIN`?